packages/gatsby/src/utils/source-nodes.ts

```diff
@@ -9,45 +9,6 @@ import type { GatsbyIterable } from "../datastore/common/iterable"
 
 const { deleteNode } = actions
 
-/**
- * Finds the name of all plugins which implement Gatsby APIs that
- * may create nodes, but which have not actually created any nodes.
- */
-function discoverPluginsWithoutNodes(
-  storeState: IGatsbyState,
-  nodes: GatsbyIterable<IGatsbyNode>
-): Array<string> {
-  // Find out which plugins own already created nodes
-  const nodeOwnerSet = new Set([`default-site-plugin`])
-  nodes.forEach(node => nodeOwnerSet.add(node.internal.owner))
-
-  return storeState.flattenedPlugins
-    .filter(
-      plugin =>
-        // "Can generate nodes"
-        plugin.nodeAPIs.includes(`sourceNodes`) &&
-        // "Has not generated nodes"
-        !nodeOwnerSet.has(plugin.name)
-    )
-    .map(plugin => plugin.name)
-}
-
-/**
- * Warn about plugins that should have created nodes but didn't.
- */
-function warnForPluginsWithoutNodes(
-  state: IGatsbyState,
-  nodes: GatsbyIterable<IGatsbyNode>
-): void {
-  const pluginsWithNoNodes = discoverPluginsWithoutNodes(state, nodes)
-
-  pluginsWithNoNodes.map(name =>
-    report.warn(
-      `The ${name} plugin has generated no Gatsby nodes. Do you need it? This could also suggest the plugin is misconfigured.`
-    )
-  )
-}
-
```

Comment:
I removed this because we're no longer iterating on every node that exists. Loading and iterating on all nodes is very expensive and doing it just for this check isn't worth it.
I also don't consider this warning very helpful. In a production build if some bug caused a plugin to not create nodes queries will certainly fail (otherwise it doesn't matter that those nodes dissapeared).
In local dev if there are no nodes you will find out very quickly when you try to query for them.

I considered adding a boolean for each plugin to redux the first time `createNode` is called for each plugin, but it just doesn't seem worth it imo. I think it's better to just remove this warning.

Reply:
I disagree, this has been useful if you've setup the wrong credentials. If queries fail it doesn't immediately indicates that it's the source plugin being broken.

You should PR that change to mark it through redux

Reply:
ok, will do :)